noir-projects/aztec-nr/aztec/src/oracle/keys.nr

```diff
@@ -22,3 +22,22 @@ fn get_public_keys_and_partial_address(address: AztecAddress) -> (PublicKeys, Pa
 
     (keys, partial_address)
 }
+
+#[oracle(getPublicKeysAndPartialAddressWithNpkMHash)]
+fn get_public_keys_and_partial_address_with_npk_m_hash_oracle(_npk_m_hash: Field) -> [Field; 9] {}
+
+unconstrained fn get_public_keys_and_partial_address_with_npk_m_hash_oracle_wrapper(npk_m_hash: Field) -> [Field; 9] {
+    get_public_keys_and_partial_address_with_npk_m_hash_oracle(npk_m_hash)
+}
+
+fn get_public_keys_and_partial_address_with_npk_m_hash(npk_m_hash: Field) -> ([GrumpkinPoint; 4], PartialAddress) {
```

Comment:
Was @benesjan not doing something similar like this for the public-keys in a struct? Collisions?

Reply:
I refactored [this oracle](https://github.com/AztecProtocol/aztec-packages/blob/e9c7e5f7a400efc746a734ab38bae6ab3d80cdad/noir-projects/aztec-nr/aztec/src/oracle/keys.nr#L11) to use the PublicKeys struct.

But yes, surprised there are no conflicts. It's suspicious.

@sklppy88 please use the PublicKeys struct here as well. My work is now on master.